test/box-luatest/gh_6506_wakeup_writing_to_wal_fiber_test.lua

```diff
@@ -0,0 +1,43 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new{alias = 'master'}
+    g.server:start()
+end
+
+g.after_all = function()
+    g.server:drop()
+end
+
+g.test_wakeup_writing_to_wal_fiber = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local fiber = require('fiber')
+
+        local build_path = os.getenv("BUILDDIR")
+        package.cpath = build_path..'/test/box-luatest/?.so;'..build_path..'/test/box-luatest/?.dylib;'..package.cpath
+        local lib = box.lib.load('gh_6506_lib')
+        local save_fiber = lib:load('save_fiber')
+        local wakeup_saved = lib:load('wakeup_saved')
+
+        local s = box.schema.create_space('test')
+        s:create_index('pk')
+
+        local f = fiber.new(function()
+            save_fiber()
+            s:replace{1}
+            fiber.yield()
+        end)
+        f:set_joinable(true)
+
+        -- Start fiber f
+        fiber.yield()
+        -- Wakeup f while it is writing to WAL
+        wakeup_saved()
+        -- Check that f did not crash
+        local st, _ = f:join()
+        t.assert_equals(st, true)
```

Comment:
You can also write `t.assert(st)`, it will be the same. `join` returns true/false as a first value. Up to you.

Reply:
Done